src/plugins/data/common/utils/helpers.ts

```diff
@@ -53,11 +58,15 @@ export const handleQueryResults = async <T>(
   } while (queryStatus !== 'SUCCESS' && queryStatus !== 'FAILED');
 
   if (queryStatus === 'FAILED') {
-    throw new Error(
+    const error = Error(
       i18n.translate('data.search.request.failed', {
         defaultMessage: 'An error occurred while executing the search query',
       })
     );
+    return {
+      status: 'failed',
+      body: { error },
```

Comment:
the returned error only contains the hard coded message "An error occurred while executing the search query", should we return the actual backend error to tell users what the error was?